nilearn/connectivity/embedding.py

```diff
@@ -0,0 +1,394 @@
+        The geometric mean of the covariance matrices.
+
+    `whitening_` : numpy.ndarray
+        The inverted square-rooted geometric mean of the covariance matrices.
+
+    Note
+    ----
+    For the use of "tangent" measure, see the paper:
+    G. Varoquaux et al. "Detection of brain functional-connectivity difference
+    in post-stroke patients using group-level covariance modeling, MICCAI 2010.
+    """
+
+    def __init__(self, cov_estimator=EmpiricalCovariance(assume_centered=True),
+                 measure='covariance'):
+        self.cov_estimator = cov_estimator
+        self.measure = measure
+
+    def fit(self, X, y=None):
+        """Fit the covariance estimator to the given time series for each
+        subject.
+
+        Parameters
+        ----------
+        X : list of numpy.ndarray, shapes (n_samples, n_features)
+            The input subjects time series.
+
+        Attributes
+        ----------
+        `cov_estimator_` : estimator object
+            A new covariance estimator with the same parameters as
+            cov_estimator.
+
+        `tangent_mean_` : numpy.ndarray
+            The mean connectivity for the tangent measure.
+
+        `whitening_` : numpy.ndarray
+            The inverted square-rooted mean for the tangent measure.
+
+        Returns
+        -------
+        self : CovEmbedding instance
+            The object itself. Useful for chaining operations.
+        """
+        self.cov_estimator_ = clone(self.cov_estimator)
+
+        if self.measure == 'tangent':
+            covs = [self.cov_estimator_.fit(x).covariance_ for x in X]
+            self.tangent_mean_ = _geometric_mean(covs, max_iter=30, tol=1e-7)
+            self.whitening_ = _map_sym(lambda x: 1. / np.sqrt(x),
+                                       self.tangent_mean_)
+
+        return self
+
+    def transform(self, X):
+        """Apply transform to covariances matrices to get the connectivity
+        coefficients for the chosen measure.
+
+        Parameters
+        ----------
+        X : list of numpy.ndarray with shapes (n_samples, n_features)
+            The input subjects time series.
+
+        Returns
+        -------
+        output : numpy.ndarray, shape (len(X), \
+                                       n_features * (n_features + 1) / 2)
+             The transformed covariance matrices.
+        """
+        covs = [self.cov_estimator_.fit(x).covariance_ for x in X]
+        covs = np.array(covs)
+        if self.measure == 'covariance':
+            pass
+        elif self.measure == 'tangent':
+            covs = [_map_sym(np.log, self.whitening_.dot(c).dot(
+                             self.whitening_)) for c in covs]
+        elif self.measure == 'precision':
+            covs = [_map_sym(lambda x: 1. / x, g) for g in covs]
+        elif self.measure == 'partial correlation':
+            covs = [_prec_to_partial(_map_sym(lambda x: 1. / x, g))
+                    for g in covs]
```

Comment:
```_map_sym``` applies the given function (x -> 1./x) to the matrix eigenvalues, keeping the eigenvectors, so actually this inverses the matrix.